### PR TITLE
Add support for Additional Conformance Test Flags

### PR DESCRIPTION
**Pull request by reetasingh**
Fixes https://github.com/kubernetes-sigs/hydrophone/issues/95

verified run by passing additional flags

```
reeta@Reetas-MacBook-Pro hydrophone % go run main.go --extra-args="--list-conformance-tests=true,--allowed-not-ready-nodes=2"
23:12:36 INF API endpoint : https://127.0.0.1:50108
23:12:36 INF Server version : version.Info{Major:"1", Minor:"24", GitVersion:"v1.24.0", GitCommit:"4ce5a8954017644c5420bae81d72b09b735c21f0", GitTreeState:"clean", BuildDate:"2022-05-19T15:42:59Z", GoVersion:"go1.18.1", Compiler:"gc", Platform:"linux/arm64"}
23:12:36 INF Using conformance image : 'registry.k8s.io/conformance:v1.24.0'
23:12:36 INF Using busybox image : 'registry.k8s.io/e2e-test-images/busybox:1.36.1-1'
23:12:36 INF Test framework will start '1' threads and use verbosity '4'
23:12:36 INF namespace created conformance

23:12:36 INF serviceaccount created conformance-serviceaccount

23:12:36 INF clusterrole created conformance-serviceaccount

23:12:36 INF clusterrolebinding created conformance-serviceaccount-role

23:12:36 INF pod created e2e-conformance-test

sep is " " args are "--list-conformance-tests=true --allowed-not-ready-nodes=2"split [--list-conformance-tests=true --allowed-not-ready-nodes=2]
2024/01/28 07:12:36 Running command:
Command env: []
Run from directory: 
Executable path: /usr/local/bin/ginkgo
Args (comma-delimited): /usr/local/bin/ginkgo,--focus=\[Conformance\],--skip=,--noColor=true,/usr/local/bin/e2e.test,--,--disable-log-dump,--repo-root=/kubernetes,--provider=skeleton,--report-dir=/tmp/results,--kubeconfig=,--list-conformance-tests=true,--allowed-not-ready-nodes=2
2024/01/28 07:12:36 Now listening for interrupts
- testname: Admission webhook, list mutating webhooks
  codename: '[sig-api-machinery] AdmissionWebhook [Privileged:ClusterAdmin] listing
    mutating webhooks should work [Conformance]'
  description: Create 10 mutating webhook configurations, all with a label. Attempt
    to list the webhook configurations matching the label; all the created webhook
    configurations MUST be present. Attempt to create an object; the object MUST be
    mutated. Attempt to remove the webhook configurations matching the label with
    deletecollection; all webhook configurations MUST be deleted. Attempt to create
    an object; the object MUST NOT be mutated.
  release: v1.16
```

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -129,6 +129,9 @@ func init() {
 	rootCmd.Flags().StringVar(&testRepo, "test-repo", "", "skip specific tests. allows regular expressions.")
 	viper.BindPFlag("test-repo", rootCmd.Flags().Lookup("test-repo"))
 
+	rootCmd.Flags().StringSlice("extra-conformance-test-args", []string{}, "Additional parameters to be provided to the conformance container. These parameters should be specified as key-value pairs, separated by commas. Each parameter should start with -- (e.g., --list-labels=true,--allowed-not-ready-nodes=2)")
```
Comment (rjsadow) [inline cmd/root.go:132]: I would prefer the flag name to just be `extra-args`. Your explanation here is great and should be clear to users that these args are passed to to the conformance pod.

File: pkg/common/args_test.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2023 The Kubernetes Authors.
```
Comment (rjsadow) [inline pkg/common/args_test.go:2]: Nit, this should be 2024

File: pkg/common/args_test.go [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+*/
+package common
+
+import (
+	"testing"
+
+	"github.com/spf13/viper"
+)
+
+func TestValidateArgs(t *testing.T) {
```
Comment (rjsadow) [inline pkg/common/args_test.go:24]: This is great, well done.

**Review by rjsadow — COMMENTED** [review]
This looks great! The test is currently failing because `make fmt` needs to be ran.

File: cmd/root.go (hunk 1 above)
Comment (reetasingh) [reply cmd/root.go]: fixed.

**Comment by reetasingh** [comment]
@rjsadow fixed review comments. I am not clear on whats wrong with the boilerplate header for args_test.go. Verify job is failing

**Comment by rjsadow** [comment]
Yeah, seems odd. I'll look more closely in a little bit.

File: pkg/common/args_test.go [hunk 4]
```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
```
Comment (rjsadow) [inline pkg/common/args_test.go:8]:
```suggestion
    http://www.apache.org/licenses/LICENSE-2.0
```

Comment (rjsadow) [reply pkg/common/args_test.go:8]: Lets give this a try. I believe the boiler plate expects exactly 4 spaces (no tabs). I'm working on some modifications to the verify scripts now to clarify the information a bit better.

Comment (reetasingh) [reply pkg/common/args_test.go]: it worked

**Comment by rakshitgondwal** [comment]
/lgtm
I believe we can improve the test coverage, but this is a good start. We can cover this as another issue.

**Pull request merged**